### PR TITLE
EZP-27008: Use UDW to browse content from discovery bar

### PR DESCRIPTION
**Pull request by StephaneDiot**
jira: https://jira.ez.no/browse/EZP-27008

## Description

Goal is to replace our old fashioned content tree by a brand new UDW explorer to be able to explore from the discovery bar. :call_me_hand: 

## Screencast 

https://youtu.be/eBAuB9TXMU4

## Tests

- Unit and manual tests

### CODE REVIEW

File: Resources/public/js/views/actions/ez-treeactionview.js [hunk 1]
```diff
@@ -22,20 +22,13 @@ YUI.add('ez-treeactionview', function (Y) {
      * @extends eZ.ButtonActionView
      */
     Y.eZ.TreeActionView = Y.Base.create('treeActionView', Y.eZ.ButtonActionView, [Y.eZ.Expandable], {
```
Comment (dpobel) [inline Resources/public/js/views/actions/ez-treeactionview.js]: looking at your patch, I'm starting to think that it does not make any sense to not deprecate this component as well. so I think, this one should also be deprecated completely and in the discovery view, it should be replaced by a raw ButtonActionView

File: Resources/public/js/views/ez-discoverybarview.js [hunk 2]
```diff
@@ -54,8 +54,8 @@ YUI.add('ez-discoverybarview', function (Y) {
                             label: Y.eZ.trans('discoverybar.search', {}, 'bar'),
                             priority: 900
                         }),
-                        new Y.eZ.TreeActionView({
-                            actionId: "tree",
+                        new Y.eZ.ButtonActionView({
+                            actionId: "find",
```
Comment (dpobel) [inline Resources/public/js/views/ez-discoverybarview.js]: `browse` would be more suited

File: Resources/public/js/views/services/ez-discoverybarviewservice.js [hunk 3]
```diff
@@ -20,10 +20,59 @@ YUI.add('ez-discoverybarviewservice', function (Y) {
...
+        _fireContentDiscover: function () {
+            var startingLocationId;
+
+            if (this._isLocationViewDisplayed()) {
+               startingLocationId = this.get('app').get('activeView').get('location').get('id');
+            }
+
+            this.fire('contentDiscover', {
+                config: {
+                    title: 'Find location', //provisional
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-discoverybarviewservice.js]: this really needs to be translated :)

File: Resources/public/js/views/services/ez-discoverybarviewservice.js [hunk 4]
```diff
@@ -20,10 +20,59 @@ YUI.add('ez-discoverybarviewservice', function (Y) {
...
+
+        /**
+         * Navigate to the selected location.
+         *
+         * @method _navigateToLocation
+         * @param {eventFacade} e
+         * @protected
+         */
+        _navigateToLocation: function (e) {
+            this.get('app').navigateTo('viewLocation',
```
Comment (dpobel) [inline Resources/public/js/views/services/ez-discoverybarviewservice.js]: cs:

```js
this.get('app').navigateTo('viewLocation', {
    id: e.selection.location.get('id'),
    languageCode: e.selection.content.get('mainLanguageCode')
});
```

File: Tests/js/views/services/assets/ez-discoverybarviewservice-tests.js [hunk 5]
```diff
@@ -88,7 +89,147 @@ YUI.add('ez-discoverybarviewservice-tests', function (Y) {
...
+
+            Y.eZ.Location = Y.Base.create('location', Y.Base, [], {}, {
+                ATTRS: {
+                    id: {
+                        value: this.locationId
+                    }
+                }
+            });
+
+            this.location = new Y.eZ.Location();
```
Comment (dpobel) [inline Tests/js/views/services/assets/ez-discoverybarviewservice-tests.js]: you should just use Y.eZ.Base (and remove the definition of Y.eZ.Location)

File: Tests/js/views/services/assets/ez-discoverybarviewservice-tests.js [hunk 6]
```diff
@@ -88,7 +89,147 @@ YUI.add('ez-discoverybarviewservice-tests', function (Y) {
...
+                    "startingLocationId should be undefined"
+                );
+            }, this));
+            this.service.fire('findAction');
+
+            Assert.isTrue(contentDiscoverFired, 'contentDiscover should be fired');
+        },
+
+        "Should navigate to the selected location when content is discovered": function () {
+            var locationMock = new Y.Mock(),
```
Comment (dpobel) [inline Tests/js/views/services/assets/ez-discoverybarviewservice-tests.js]: using mocks here does not bring much, you should also use a Y.eZ.Base

File: Tests/js/views/services/assets/ez-discoverybarviewservice-tests.js [hunk 7]
```diff
@@ -88,7 +89,147 @@ YUI.add('ez-discoverybarviewservice-tests', function (Y) {
...
+                method: 'get',
+                args: ["activeView"],
+                callCount: 2,
+                returns: this.activeView
+            });
+
+            Y.Mock.expect(locationMock, {
+                method: 'get',
+                args: ['id'],
+                returns: this.locationId
```
Comment (dpobel) [inline Tests/js/views/services/assets/ez-discoverybarviewservice-tests.js]: you use the same location id for both the starting Location and the picked Location

**Comment by StephaneDiot** [comment]
updated @dpobel

**Pull request closed without merging**